src/igvfd/audit/file_set.py

```diff
@@ -21,3 +21,128 @@ def audit_no_files(value, system):
             f'has no files.'
         )
         yield AuditFailure('missing files', detail, level='WARNING')
+
+
+@audit_checker('FileSet', frame='object')
+def audit_missing_seqspec(value, system):
+    '''
+        audit_detail: Sequence files in a file set are expected to link to a seqspec file.
+        audit_category: missing sequence specification
+        audit_levels: ERROR
+    '''
+    if 'files' in value:
+        no_seqspec = []
+        for file in value['files']:
+            if file.startswith('/sequence-files/'):
+                sequence_file_object = system.get('request').embed(file, '@@object?skip_calculated=true')
+                if not sequence_file_object.get('seqspec'):
+                    no_seqspec.append(file)
+        if no_seqspec:
+            no_seqspec = ', '.join([audit_link(path_to_text(file_no_seqspec), file_no_seqspec)
+                                   for file_no_seqspec in no_seqspec])
+            detail = (
+                f'File set {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file(s): '
+                f'{no_seqspec} which do not have a seqspec configuration file.'
+            )
+            yield AuditFailure('missing sequence specification', detail, level='ERROR')
+
+
+@audit_checker('FileSet', frame='object')
+def audit_missing_seqspec_files(value, system):
+    '''
+        audit_detail: All files associated with a seqspec (both sequencing and seqspec files) are expected to be linked on the same file set.
+        audit_category: missing related files
```

Comment:
I am not sure it is missing? maybe you have agreed with Cricket on this strategy? and I dont want to interfere if you did.

Reply:
Would they not be missing? A sequence file and its seqspec would always be expected to belong to the same file set, so if one is not linked there it would be missing.

Reply:
it could be correct, but living in the other set, but you can call it missing